lib/Data/Database.php

```diff
@@ -535,19 +525,36 @@ protected function _getExpiredPastes($batchsize)
         return $pastes;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function getAllPastes()
+    {
+        $pastes = array();
+        $rows   = $this->_select(
+            'SELECT "dataid" FROM "' . $this->_sanitizeIdentifier('paste') . '"',
+            array()
+        );
+        if (is_array($rows) && count($rows)) {
+            foreach ($rows as $row) {
+                $pastes[] = $row['dataid'];
+            }
+        }
```

Comment:
The `_select()` function does some extra bits for the OCI driver for converting CLOB type columns from streams to strings, but the dataid will be a `CHAR(16)` in all drivers, so we could get away just fetching only the one row, straight:
```suggestion
        $pastes = $this->_db->query(
            'SELECT "dataid" FROM "' . $this->_sanitizeIdentifier('paste') . '"'
        )->fetchAll(PDO::FETCH_COLUMN, 0);
```
... and I should apply the same change to `_getExpiredPastes()`.